package.xml.in

```diff
@@ -1,7 +1,7 @@
 <?xml version="1.0"?>
 <package>
-  <name>serial</name>
-  <version>1.2.1</version>
+  <name>@PROJECT_NAME@</name>
+  <version>@PROJECT_VERSION@</version>
```

Comment:
Does rosdep support `package.xml` -> `package.xml.in` when using rosdep to install all dependencies?

Reply:
Not that I've been able to. If the project name is constant, then the maintainer just needs to keep this in sync with the CMakeLists, which is what most of the other packages do anyways.

Reply:
As I understand this at a glance, rosdep is not aware of package.xml.in , it only uses package.xml which is updated at each cmake invocation

> If the project name is constant, then the maintainer just needs to keep this in sync with the CMakeLists, which is what most of the other packages do anyways.

Indeed. However, the version is not constant.

Reply:
You can't do this, sorry. The `package.xml` needs to exist before building.

Reply:
It complicates the upstream maintenance, having  especially the version in both CMakeLists and package.xml is not optimal.

It could be done the other way around: CMakeLists reads the version from package.xml.  That would also be a consistent setup.

However, from a packaging perspective it doesn't really matter, so let's drop this for now

Reply:
> 
> It could be done the other way around: CMakeLists reads the version from package.xml.  That would also be a consistent setup.
> 


Dart does exactly this: https://github.com/dartsim/dart/blob/main/CMakeLists.txt#L29 .

Reply:
Yes. Even better would be to scan package.xml before the cmake `project()` invocation.  If done this way the version could be set in `project(...)`; cmake has some nice provisions to set up standard variables from this.